api/org/commcare/api/json/PromptToJson.java

```diff
@@ -0,0 +1,224 @@
+                return;
+            case Constants.DATATYPE_LONG:
+            case Constants.DATATYPE_DECIMAL:
+                obj.put("answer", (double) answerValue.getValue());
+                return;
+            case Constants.DATATYPE_DATE:
+                obj.put("answer", (DateUtils.formatDate((Date) answerValue.getValue(), DateUtils.FORMAT_ISO8601)));
+                return;
+            case Constants.DATATYPE_TIME:
+                obj.put("answer", answerValue.getDisplayText());
+                return;
+            case Constants.DATATYPE_DATE_TIME:
+                obj.put("answer", ((Date) answerValue.getValue()).getTime());
+                return;
+            case Constants.DATATYPE_CHOICE:
+                Selection singleSelection = ((Selection) answerValue.getValue());
+                singleSelection.attachChoice(prompt.getQuestion());
+                obj.put("answer", ((Selection) answerValue.getValue()).getTouchformsIndex());
+                return;
+            case Constants.DATATYPE_CHOICE_LIST:
+                Vector<Selection> selections = ((SelectMultiData) answerValue).getValue();
+                JSONArray acc = new JSONArray();
+                for (Selection selection : selections) {
+                    selection.attachChoice(prompt.getQuestion());
+                    int ordinal = selection.getTouchformsIndex();
+                    acc.put(ordinal);
+                }
+                obj.put("answer", acc);
+                return;
+            case Constants.DATATYPE_GEOPOINT:
+                GeoPointData geoPointData = ((GeoPointData) prompt.getAnswerValue());
+                double[] coords = new double[]{geoPointData.getLatitude(), geoPointData.getLongitude()};
+                obj.put("answer", coords);
+                return;
+        }
+    }
+
+    /**
+     * Given a prompt, generate a JSONArray of the possible select choices. return empty array if no choices.
+     */
+    private static JSONArray parseSelect(FormEntryPrompt prompt) {
+        JSONArray obj = new JSONArray();
+        for (SelectChoice choice : prompt.getSelectChoices()) {
+            obj.put(prompt.getSelectChoiceText(choice));
+        }
+        return obj;
+    }
+
+    //TODO WSP: What the fuck is drew doing XFormPlayer parse_style_info
+    // https://github.com/dimagi/touchforms/blob/master/touchforms/backend/xformplayer.py#L400
+    private static JSONObject parseStyle(FormEntryPrompt prompt) {
+        String hint = prompt.getAppearanceHint();
+        if (hint == null) {
+            return null;
+        }
+        JSONObject ret = new JSONObject().put("raw", hint);
+        return ret;
+    }
+
+
+    private static String parseControlType(FormEntryPrompt prompt) {
+        if (prompt.getControlType() == Constants.CONTROL_TRIGGER) {
+            return "info";
+        }
+        switch (prompt.getDataType()) {
+            case Constants.DATATYPE_NULL:
```

Comment:
Can you add a comment describing why this switch case is falling through to the next? This is a common gotcha, so it is worth explicitly asserting this is what we want.